### PR TITLE
Adapt install script to not require cluster name as mandatory parameter

### PR DESCRIPTION
**Pull request by DTMad**

### CODE REVIEW

File: deploy/install.sh [hunk 1]
```diff
@@ -93,7 +94,32 @@ applyDynatraceOperator() {
 }
 
 applyDynaKubeCR() {
-  cat <<EOF | "${CLI}" apply -f -
+  if [ -n "$CLUSTER_NAME" ]; then
```
Comment (lrgar) [inline deploy/install.sh:97]: If I'm not mistaken, `-n` means non-empty, and this example doesn't use `CLUSTER_NAME`, so perhaps it's the other way around?

File: deploy/install.sh [hunk 2]
```diff
@@ -142,8 +169,33 @@ addK8sConfiguration() {
     exit 1
   fi
 
-  json="$(
-    cat <<EOF
+  if [ -n "$CLUSTER_NAME" ]; then
```
Comment (lrgar) [inline deploy/install.sh:172]: (Kind of) Same as above.

File: deploy/install.sh [hunk 3]
```diff
@@ -142,8 +169,33 @@ addK8sConfiguration() {
...
+  "eventsFieldSelectors": [
+    {
+      "label": "Node events",
+      "fieldSelector": "involvedObject.kind=Node",
+      "active": true
+    }
+  ],
+  "workloadIntegrationEnabled": true,
+  "eventsIntegrationEnabled": false,
+  "activeGateGroup": "${CLUSTER_NAME}",
```
Comment (lrgar) [inline deploy/install.sh:189]: I presume you don't want to use this field here, and only on the else.

File: deploy/install.sh [hunk 4]
```diff
@@ -162,22 +214,28 @@ addK8sConfiguration() {
...
+  fi
 
   response=$(apiRequest "POST" "/config/v1/kubernetes/credentials" "${json}")
 
-  if echo "$response" | grep "${CLUSTER_NAME}" >/dev/null 2>&1; then
-    echo "Kubernetes monitoring successfully setup."
-  else
-    echo "Error adding Kubernetes cluster to Dynatrace: $response"
+  if [ -n "$CLUSTER_NAME" ]; then
+    if echo "$response" | grep "${CLUSTER_NAME}" >/dev/null 2>&1; then
```
Comment (lrgar) [inline deploy/install.sh:223]: So no checks if connection name is used?

File: deploy/install.sh [hunk 5]
```diff
@@ -162,22 +214,28 @@ addK8sConfiguration() {
...
   fi
 }
 
 checkForExistingCluster() {
   response=$(apiRequest "GET" "/config/v1/kubernetes/credentials" "")
-  if echo "$response" | grep -FEq "\"name\":\"${CLUSTER_NAME}\""; then
-    echo "Error: Cluster name already exists!"
-    exit 1
+
+  if [ -n "$CLUSTER_NAME" ]; then
```
Comment (lrgar) [inline deploy/install.sh:234]: Technically you can use the connection name instead here to verify.

**Review by lrgar — CHANGES_REQUESTED** [review]

File: deploy/install.sh [hunk 6]
```diff
@@ -66,6 +74,9 @@ if [ -z "$CLUSTER_NAME" ]; then
     echo "Error: cluster name too long!"
     exit 1
   fi
+  CONNECTION_NAME=$CLUSTER_NAME
```
Comment (lrgar) [inline deploy/install.sh:77]:
```suggestion
  CONNECTION_NAME="$CLUSTER_NAME"
```

File: deploy/install.sh [hunk 7]
```diff
@@ -142,8 +174,32 @@ addK8sConfiguration() {
     exit 1
   fi
 
-  json="$(
-    cat <<EOF
+  if [ -z "$CLUSTER_NAME" ]; then
+    CONNECTION_NAME="$(echo "${K8S_ENDPOINT}" | awk -F[/:] '{print $4}')"
+
```
Comment (lrgar) [inline deploy/install.sh:179]: No need to set it.
```suggestion
```

**Review by lrgar — APPROVED** [review]

File: deploy/install.sh [hunk 8]
```diff
@@ -175,7 +230,8 @@ EOF
 
 checkForExistingCluster() {
   response=$(apiRequest "GET" "/config/v1/kubernetes/credentials" "")
-  if echo "$response" | grep -FEq "\"name\":\"${CLUSTER_NAME}\""; then
+
+  if echo "$response" | grep -FEq "\"name\":\"${CONNECTION_NAME}\""; then
```
Comment (baichinger) [inline deploy/install.sh:234]: Causes `grep: conflicting matchers specified` for GNU grep 3.6.

**Review by baichinger — CHANGES_REQUESTED** [review]

**Review by lrgar — APPROVED** [review]

**Review by baichinger — APPROVED** [review]

**Pull request merged**